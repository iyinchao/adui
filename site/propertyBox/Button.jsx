import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Switch, Button } from "../../components"
import { SIZES, INTENTS } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    disabled: false,
    href: null,
    intent: "normal",
    leftIcon: "file-outlined",
    loading: false,
    rightIcon: "arrow-forward",
    size: "small",
    theme: null,
  }

  render() {
    const {
      disabled,
      href,
      intent,
      leftIcon,
      loading,
      rightIcon,
      size,
      theme,
    } = this.state

    const codes = `<Button
  ${disabled ? "disabled" : ""}
  ${href ? `href="${href}"` : ""}
  ${intent === "normal" ? "" : `intent="${intent}"`}
  ${leftIcon === null ? "" : `leftIcon="${leftIcon}"`}
  ${loading ? "loading" : ""}
  ${rightIcon === null ? "" : `rightIcon="${rightIcon}"`}
  ${size === "small" ? "" : `size="${size}"`}
  ${theme ? `theme="${theme}"` : ""}
>
  点击按钮
</Button>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <Button {...this.state}>点击按钮</Button>
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <dt className={styles.controlTitle}>倾向</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {INTENTS.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === intent}
                      onClick={() => this.setState({ intent: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="左图标"
                  unCheckedText="左图标"
                  checked={!!leftIcon}
                  onChange={(param) =>
                    this.setState({ leftIcon: param ? "file-outlined" : null })
                  }
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="右图标"
                  unCheckedText="右图标"
                  checked={!!rightIcon}
                  onChange={(param) =>
                    this.setState({ rightIcon: param ? "arrow-forward" : null })
                  }
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="加载态"
                  unCheckedText="加载态"
                  checked={loading}
                  onChange={(param) => this.setState({ loading: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="禁用态"
                  unCheckedText="禁用态"
                  checked={disabled}
                  onChange={(param) => this.setState({ disabled: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="轻量风格"
                  unCheckedText="轻量风格"
                  checked={!!theme}
                  onChange={(param) =>
                    this.setState({ theme: param ? "light" : null })
                  }
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="跳转链接"
                  unCheckedText="跳转链接"
                  checked={!!href}
                  onChange={(param) =>
                    this.setState({
                      href: param ? "https://ad.weixin.qq.com" : null,
                    })
                  }
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
