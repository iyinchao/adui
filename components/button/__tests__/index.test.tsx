import * as React from "react"
import TestRenderer from "react-test-renderer"
import Button from ".."

describe("Button", () => {
  it("是否正确渲染", () => {
    const wrapper = TestRenderer.create(
      <Button leftIcon="add" rightIcon="tick">
        按钮
      </Button>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("是否支持 link button", () => {
    const wrapper = TestRenderer.create(
      <Button target="_blank" href="https://ad.weixin.qq.com">
        link button
      </Button>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
