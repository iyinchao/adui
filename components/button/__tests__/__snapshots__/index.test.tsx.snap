// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 是否支持 link button 1`] = `
<a
  className="adui-button-base adui-button-normal adui-button-small"
  disabled={false}
  href="https://ad.weixin.qq.com"
  onClick={[Function]}
  target="_blank"
>
  <span
    className="adui-button-content"
  >
    link button
  </span>
</a>
`;

exports[`Button 是否正确渲染 1`] = `
<button
  className="adui-button-base adui-button-normal adui-button-small adui-button-hasLeftIcon adui-button-hasRightIcon"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <svg
    className="adui-button-leftIcon adui-icon-base"
    data-icon="add"
    data-interactive={false}
    fill="var(--gray-700)"
    height={18}
    onClick={[Function]}
    style={Object {}}
    viewBox="0 0 18 18"
    width={18}
  >
    <path
      d="M8.25 14.5C8.25 14.7761 8.47386 15 8.75 15H9.25C9.52614 15 9.75 14.7761 9.75 14.5V9.75H14.5C14.7761 9.75 15 9.52614 15 9.25V8.75C15 8.47386 14.7761 8.25 14.5 8.25H9.75V3.5C9.75 3.22386 9.52614 3 9.25 3H8.75C8.47386 3 8.25 3.22386 8.25 3.5V8.25H3.5C3.22386 8.25 3 8.47386 3 8.75V9.25C3 9.52614 3.22386 9.75 3.5 9.75H8.25V14.5Z"
      fillRule="evenodd"
    />
  </svg>
  <span
    className="adui-button-content"
  >
    按钮
  </span>
  <svg
    className="adui-button-rightIcon adui-icon-base"
    data-icon="tick"
    data-interactive={false}
    fill="var(--gray-700)"
    height={18}
    onClick={[Function]}
    style={Object {}}
    viewBox="0 0 18 18"
    width={18}
  >
    <path
      d="M15.0962 6.43932C15.2914 6.24406 15.2914 5.92748 15.0962 5.73222L14.3891 5.02511C14.1938 4.82985 13.8772 4.82985 13.682 5.02511L7.67157 11.0355L4.48958 7.85354C4.29432 7.65827 3.97774 7.65827 3.78248 7.85354L3.07537 8.56064C2.88011 8.7559 2.88011 9.07249 3.07537 9.26775L7.31801 13.5104C7.51328 13.7057 7.82986 13.7057 8.02512 13.5104L15.0962 6.43932Z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
